worlds/cvcotm/client.py

```diff
@@ -0,0 +1,551 @@
+                        if flag_id == FLAG_HIT_IRON_MAIDEN_SWITCH and ctx.slot_data["iron_maiden_behavior"] == \
+                                IronMaidenBehavior.option_vanilla:
+                            locs_to_send.add(cvcotm_location_info[lname.ct21].code + BASE_ID)
+
+                        # If the DSS tutorial flag is set, let the client know, so it's not shown again for
+                        # subsequently-received cards.
+                        if flag_id == FLAG_SAW_DSS_TUTORIAL:
+                            self.saw_dss_tutorial = True
+
+                        if flag_id in EVENT_FLAG_MAP:
+                            checked_set_events[EVENT_FLAG_MAP[flag_id]] = True
+
+                            # Update the client's statuses for the Battle Arena and Dracula goals.
+                            if flag_id == FLAG_WON_BATTLE_ARENA:
+                                self.won_battle_arena = True
+
+                            if flag_id == FLAG_DEFEATED_DRACULA_II:
+                                self.killed_dracula_2 = True
+
+            # Send Locations if there are any to send.
+            if locs_to_send != self.local_checked_locations:
+                self.local_checked_locations = locs_to_send
+
+                if locs_to_send is not None:
+                    # Capture all the Locations with non-local Items to send that are in ctx.missing_locations
+                    # (the ones that were definitely never sent before).
+                    self.sent_message_queue += [loc for loc in locs_to_send if loc in ctx.missing_locations and
+                                                ctx.locations_info[loc].player != ctx.slot]
+
+                    await ctx.send_msgs([{
+                        "cmd": "LocationChecks",
+                        "locations": list(locs_to_send)
+                    }])
+
+            # Check the win condition depending on what our completion goal is.
+            # The Dracula option requires the "killed Dracula II" flag to be set or being in the credits state.
+            # The Battle Arena option requires the Shinning Armor pickup flag to be set.
+            # Otherwise, the Battle Arena and Dracula option requires both of the above to be satisfied simultaneously.
+            if ctx.slot_data["completion_goal"] == CompletionGoal.option_dracula:
+                win_condition = self.killed_dracula_2
+            elif ctx.slot_data["completion_goal"] == CompletionGoal.option_battle_arena:
+                win_condition = self.won_battle_arena
+            else:
+                win_condition = self.killed_dracula_2 and self.won_battle_arena
+
+            # Send game clear if we've satisfied the win condition.
+            if not ctx.finished_game and win_condition:
+                await ctx.send_msgs([{
+                    "cmd": "StatusUpdate",
+                    "status": ClientStatus.CLIENT_GOAL
+                }])
```

Comment:
I think this will just send finished game every time.
You need to set `ctx.finished_game = True` yourself, that's not done for you.

I think I would ask that you play another seed with a local MultiServer.py using the `--log_network` arg and try to look out for packet spam. If you can somehow arbitrarily slow it down, that would be great too, idk how to do that though.